internal/registry/registry_test.go

```diff
@@ -433,15 +434,15 @@ func TestPollProvider(t *testing.T) {
 	select {
 	case pinfo := <-r.SyncChan():
 		require.Equal(t, pinfo.AddrInfo.ID, peerID, "unexpected provider info on sync channel, expected %q got %q", peerID.String(), pinfo.AddrInfo.ID.String())
-		require.True(t, pinfo.Inactive(), "Expected provider to be marked inactive")
+		require.False(t, pinfo.Inactive(), "Expected provider to not be marked inactive")
```

Comment:
I think this means that the test that a provider becomes inactive in now missing.  Instead of changing the lin above, I think that `deactivateAfter` shold be set to a value that makes the provider become inactive and the original test condition to be met.